samtranslator/model/api/api_generator.py

```diff
@@ -108,6 +109,8 @@ def _construct_rest_api(self):
 
         self._add_cors()
         self._add_auth()
+        # Post Process OpenApi Auth Settings
+        self.definition_body = self._openapi_auth_postprocess(self.definition_body)
```

Comment:
@praneetap In order to support the ApiKeyRequired setting at the Function level (without requiring an explicit setting on the API), I had to move the `_openapi_auth_postprocess` method call up a level to ensure it was called on every template reqardless of if there's an explicit API Auth setting.  If you have another suggestion I'm happy to change.